runners/spark/job-server/container/Dockerfile

```diff
@@ -16,7 +16,7 @@
 # limitations under the License.
 ###############################################################################
 
-FROM openjdk:8
+FROM openjdk:16.0.2
```

Comment:
This is a breaking change, we can't perform this upgrade without having a larger conversation about it. If you think we need to do this, please send an email to `dev@apache.org` proposing the change